### PR TITLE
Serf deprecation mode

### PR DESCRIPTION
**Pull request by a-palchikov**
This PR updates the serf to reflect the changes in the agent status API. The 9.x version is still considered a bridge and even though the serf is gone as a service/binary the older 9 clusters still need to be supported for the upgrade.

Requires https://github.com/gravitational/satellite/pull/303.

### CODE REVIEW

**Review by bernardjkim — APPROVED** [review]

**Review by wadells — APPROVED** [review]
LGTM.  Please update the dependencies before merge.

**Comment by a-palchikov** [comment]
> LGTM. Please update the dependencies before merge.

Sure, once the satellite PR has been merged - sorry, I don't have write access to that repository.

**Pull request merged**
